cmd/resources/resources.go

```diff
@@ -0,0 +1,145 @@
+package resources
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"ldcli/cmd/cliflags"
+	"ldcli/cmd/validators"
+)
+
+func NewResourceCmd(parentCmd *cobra.Command, resourceName, shortDescription, longDescription string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   resourceName,
+		Short: shortDescription,
+		Long:  longDescription,
+		//TODO: add tracking here
+	}
+
+	parentCmd.AddCommand(cmd)
+
+	return cmd
+}
+
+type OperationData struct {
+	Short        string
+	Long         string
+	Use          string
+	Params       []Param
+	HTTPMethod   string
+	RequiresBody bool
+	Path         string
+}
+
+type Param struct {
+	Name        string
+	In          string
+	Description string
+	Type        string
+}
+
+type OperationCmd struct {
+	OperationData
+	client *http.Client
+	cmd    *cobra.Command
+}
+
+func (op *OperationCmd) initFlags() error {
+	if op.RequiresBody {
+		op.cmd.Flags().StringP(cliflags.DataFlag, "d", "", "Input data in JSON")
+		err := op.cmd.MarkFlagRequired(cliflags.DataFlag)
+		if err != nil {
+			return err
+		}
+		err = viper.BindPFlag(cliflags.DataFlag, op.cmd.Flags().Lookup(cliflags.DataFlag))
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, p := range op.Params {
+		shorthand := fmt.Sprintf(p.Name[0:1]) // todo: how do we handle potential dupes
```

Comment:
oh i meant duplicate shorthands

Reply:
we can see if that actually ends up being an issue once we get the rest in there maybe.

Reply:
Oh, that's interesting. It looks like it will panic with
```
panic: unable to redefine 'e' shorthand in "get" flagset: it's already used for "expand" flag
```
I suppose we could iterate through all the existing flags to check if it's already been set and then set it to something else (`e2`?), or we just don't set the shorthand.

Reply:
ya or add them to a map and check before adding it. was thinking it could be the first two letters if the one letter was already taken (but even then that's not a guaranteed unique value)

Reply:
```
e
ex
exp
expa
expan
expand
```